action.yaml

```diff
@@ -0,0 +1,149 @@
+        ### pytest-sentry configuration ###
+        if [ "$GITHUB_REPOSITORY" = "getsentry/self-hosted" ]; then
+          echo "PYTEST_SENTRY_DSN=$SELF_HOSTED_TESTING_DSN" >> $GITHUB_ENV
+          echo "PYTEST_SENTRY_TRACES_SAMPLE_RATE=0" >> $GITHUB_ENV
+
+          # This records failures on master to sentry in order to detect flakey tests, as it's
+          # expected that people have failing tests on their PRs
+          if [ "$GITHUB_REF" = "refs/heads/master" ]; then
+            echo "PYTEST_SENTRY_ALWAYS_REPORT=1" >> $GITHUB_ENV
+          fi
+        fi
+
+    - name: Get Compose
+      env:
+        COMPOSE_PATH: /usr/local/lib/docker/cli-plugins
+        COMPOSE_VERSION: "v2.26.0"
+      shell: bash
+      run: |
+        # Always remove `docker compose` support as that's the newer version
+        # and comes installed by default nowadays.
+        sudo rm -f "/usr/local/lib/docker/cli-plugins/docker-compose"
+        # Docker Compose v1 is installed here, remove it
+        sudo rm -f "/usr/local/bin/docker-compose"
+        sudo rm -f "${{ env.COMPOSE_PATH }}/docker-compose"
+        sudo mkdir -p "${{ env.COMPOSE_PATH }}"
+        sudo curl -L https://github.com/docker/compose/releases/download/${{ env.COMPOSE_VERSION }}/docker-compose-`uname -s`-`uname -m` -o "${{ env.COMPOSE_PATH }}/docker-compose"
+        sudo chmod +x "${{ env.COMPOSE_PATH }}/docker-compose"
+
+    - name: Prepare Docker Volume Caching
+      id: cache_key
+      shell: bash
+      run: |
+        # Set permissions for docker volumes so we can cache and restore
+        sudo chmod o+x /var/lib/docker
+        sudo chmod -R o+rwx /var/lib/docker/volumes
+        source .env
+        SENTRY_IMAGE_SHA=$(docker buildx imagetools inspect $SENTRY_IMAGE --format "{{println .Manifest.Digest}}")
+        echo "SENTRY_IMAGE_SHA=$SENTRY_IMAGE_SHA" >> $GITHUB_OUTPUT
+        SNUBA_IMAGE_SHA=$(docker buildx imagetools inspect $SNUBA_IMAGE --format "{{println .Manifest.Digest}}")
+        echo "SNUBA_IMAGE_SHA=$SNUBA_IMAGE_SHA" >> $GITHUB_OUTPUT
+
+    - name: Restore DB Volumes Cache
+      id: restore_cache
+      uses: actions/cache/restore@v4
+      with:
+        key: db-volumes-v4-${{ steps.cache_key.outputs.SENTRY_IMAGE_SHA }}-${{ steps.cache_key.outputs.SNUBA_IMAGE_SHA }}
+        restore-keys: |
+          db-volumes-v4-${{ steps.cache_key.outputs.SENTRY_IMAGE_SHA }}
+          db-volumes-v4-
+        path: |
+          /var/lib/docker/volumes/sentry-postgres/_data
+          /var/lib/docker/volumes/sentry-clickhouse/_data
+          /var/lib/docker/volumes/sentry-kafka/_data
+
+    - name: Install self-hosted
+      env:
+        SKIP_DB_MIGRATIONS: ${{ steps.restore_cache.outputs.cache-hit == 'true' && '1' || '' }}
+      shell: bash
+      run: |
+        # This is for the cache restore on Kafka to work in older releases
+        docker run --rm -v "sentry-kafka:/data" busybox chown -R 1000:1000 /data
+        # Add some customizations to test that path
```

Comment:
Is the goal here to test customizations as default for these tests? Currently the behavior is different so wanted to hear your thoughts here.

Reply:
Yup. Just removing the double work and doing it with customizations at all times.